docs/REVIEWING.md

```diff
@@ -73,6 +73,17 @@ Documentation is incredibly important to Vector; it is a feature and
 differentiator for Vector. Pull requests should not be merged without adequate
 documentation, nor should they be merged with "TODOs" opened for documentation.
 
+Ideally all modules should have module level documentation. Module level
+documentation can be omitted for modules where the purpose is obvious. For
+example the with the sinks there is typically a standard number of modules
+included (service.rs, sink.rs, request_builder.rs), these modules don't need
+documentation. All `pub` and `pub(crate)` functions, structs and macros must
+have documentation. Consider including examples for components that will be
+well used throughout Vector.
```

Comment:
Components here are user facing or developer facing?

Reply:
Yep, I figured as much - realistically they're going to begin overlapping as well, with the `configurable` trait driving user facing docs eventually. So I'm not entirely sure we need to break them apart.

Just wanted to clarify for myself here. I'd _love_ better documentation around some of our more complex internals.